cms/core/blocks/embeddable.py

```diff
@@ -98,3 +100,86 @@ def clean(self, value: "StructValue") -> "StructValue":
     class Meta:
         icon = "code"
         template = "templates/components/streamfield/ons_embed_block.html"
+
+
+class VideoEmbedBlock(blocks.StructBlock):
+    """A video embed block."""
+
+    link_url = blocks.URLBlock(
+        help_text=_(
+            "The URL to the video hosted on YouTube or Vimeo, for example, "
+            "https://www.youtube.com/watch?v={ video ID } or https://vimeo.com/video/{ video ID }. "
+            "Used to link to the video when cookies are not enabled."
+        )
+    )
+    image = ImageChooserBlock(help_text=_("The video cover image, used when cookies are not enabled."))
+    title = blocks.CharBlock(help_text=_("The descriptive title for the video used by screen readers."))
+    link_text = blocks.CharBlock(
+        help_text=_("The text to be shown when cookies are not enabled e.g. 'Watch the {title} on Youtube'.")
+    )
+
+    def get_context(self, value: "StreamValue", parent_context: dict | None = None) -> dict:
+        """Get the embed URL for the video based on the link URL."""
+        context: dict = super().get_context(value, parent_context=parent_context)
+        embed_url = ""
+        # Vimeo
+        if urlparse(value["link_url"]).hostname in ["www.vimeo.com", "vimeo.com", "player.vimeo.com"]:
+            url_path = urlparse(value["link_url"]).path.strip("/")
+            # Handle different Vimeo URL patterns
+            if "video/" in url_path:
```

Comment:
Ruff doesn't like this and suggests changing it as follows

Use ternary operator `video_id = url_path.split("video/")[-1] if "video/" in url_path else url_path.split("/")[0]` instead of `if`-`else`-block

However I'd argue what is written is a lot clearer. Any thoughts @zerolab?

Reply:
you can do `# noqa: SIM001` (or whatever that check is to keep the current version. and in this case I agree we should

Reply:
sorted, thanks.